http/org.wso2.carbon.transport.http.netty/src/main/java/org/wso2/carbon/transport/http/netty/contractimpl/HttpClientConnectorImpl.java

```diff
@@ -75,16 +82,25 @@ public HttpResponseFuture send(HTTPCarbonMessage httpCarbonRequest) {
         try {
             final HttpRoute route = getTargetRoute(httpCarbonRequest);
             TargetChannel targetChannel = connectionManager
-                    .borrowTargetChannel(route, srcHandler, sslConfig, httpTraceLogEnabled);
+                    .borrowTargetChannel(route, srcHandler, sslConfig, httpTraceLogEnabled, followRedirect,
+                            maxRedirectCount);
             targetChannel.getChannelFuture()
                     .addListener(new ChannelFutureListener() {
                         @Override
                         public void operationComplete(ChannelFuture channelFuture) throws Exception {
                             if (isValidateChannel(channelFuture)) {
                                 targetChannel.setChannel(channelFuture.channel());
-
+                                channelFuture.channel().attr(Constants.ORIGINAL_REQUEST).set(httpCarbonRequest);
```

Comment:
Do we need to do this for each and every request ?

Reply:
Channel attributes are set only when the redirect is on.